terraform/gcp/README.md

```diff
@@ -0,0 +1,253 @@
+* Creating a new GCP project, along with a Terraform Service Account.
+* Building a golden image using Packer.
+* Deploying a cluster with Terraform.
+
+## Install HashiCorp Tools
+
+### Nomad
+
+Download the latest version of [Nomad](https://www.nomadproject.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/nomad/0.12.0/nomad_0.12.0_linux_amd64.zip" -o nomad.zip
+unzip nomad.zip
+sudo mv nomad /usr/local/bin
+nomad --version
+```
+
+### Consul
+
+Download the latest version of [Consul](https://www.consul.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/consul/1.8.0/consul_1.8.0_linux_amd64.zip" -o consul.zip
+unzip consul.zip
+sudo mv consul /usr/local/bin
+consul --version
+```
+
+### Vault
+
+Download the latest version of [Vault](https://www.vaultproject.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/vault/1.4.3/vault_1.4.3_linux_amd64.zip" -o vault.zip
+unzip vault.zip
+sudo mv vault /usr/local/bin
+vault --version
+```
+
+### Packer
+
+Download the latest version of [Packer](https://www.packer.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/packer/1.6.0/packer_1.6.0_linux_amd64.zip" -o packer.zip
+unzip packer.zip
+sudo mv packer /usr/local/bin
+packer --version
+```
+
+### Terraform
+
+Download the latest version of [Terraform](https://www.terraform.io/) from HashiCorp's website by copying and pasting this snippet in the terminal:
+
+```console
+curl "https://releases.hashicorp.com/terraform/0.12.28/terraform_0.12.28_linux_amd64.zip" -o terraform.zip
+unzip terraform.zip
+sudo mv terraform /usr/local/bin
+terraform --version
+```
+
+### Install the GCP SDK Command Line Tools
+
+> **Note**: if you are using the free [Google Cloud Shell](https://cloud.google.com/shell) VM, you will already have `gcloud` installed. So, you can safely skip this step.
+
+To install the GCP SDK Command Line Tools, follow the installation instructions for your specific operating system: 
+
+* [Linux](https://cloud.google.com/sdk/docs/downloads-interactive#linux)
+* [MacOS](https://cloud.google.com/sdk/docs/downloads-interactive#mac)
+* [Windows](https://cloud.google.com/sdk/docs/downloads-interactive#windows)
+
+#### Initialize the SDK
+
+To perform common setup tasks like authorizing with GCP using `gcloud`, if you haven't already done so, run the following command:
+
+```console
+gcloud init
+```
+
+## Create a New Project
+
+Generate a project ID with the following command:
+
+```console
+export GOOGLE_PROJECT="nomad-gcp-$(cat /dev/random | head -c 5 | xxd -p)"
+```
+
+Using that project ID, create a new GCP [project](https://cloud.google.com/docs/overview#projects):
+
+```console
+gcloud projects create $GOOGLE_PROJECT
```

Comment:
This was also done by `gcloud init` and produces an error if done again.

Reply:
🤔 So when you optionally create a project with `gcloud init`, there is an error when creating a new project in the next steps of the tutorial?

The `gcloud init` step is only meant to be run for users first installing the SDK on their own systems, not for users already in the Cloud Shell VM where I believe it should already be installed.

Reply:
Thanks for reporting this; it would only impact a new gcloud setup!  However, this should not have produced an error, even after `gcloud init` had been run, unless you were trying to create a project that already existed. 

When initing a new setup, I received a prompt that looks like this (mileage will vary slightly for brand new GCP accounts):

```
This account has a lot of projects! Listing them all can take a while.
 [1] Enter a project ID
 [2] Create a new project
 [3] List projects
Please enter your numeric choice:  2

Enter a Project ID. Note that a Project ID CANNOT be changed later.
Project IDs must be 6-30 characters (lowercase ASCII, digits, or
hyphens) in length and start with a lowercase letter. 
```

As of

```
gcloud --version
Google Cloud SDK 301.0.0
alpha 2020.07.10
beta 2020.07.10
bq 2.0.58
core 2020.07.10
gsutil 4.51
kubectl 2020.05.01
```

gcloud init seemed to accept a `--project` argument to bypass the interactive prompt, but still did not create a project. So you could pass it 

```
gcloud init --project plurfleblaaah
gcloud projects create plurfleblaaah
```

Reply:
🤔 I don't know if I like the `--project` flag passing to `gcloud init`, but I don't _necessarily_ hate it either.

I think we _just_ need the user to be authenticated with GCP, so replacing the initialization step with an authentication step using `gcloud auth login` _feels_ like the simplest solution. I think it even might make more sense conceptually for new users?

Reply:
Indeed. And cloud shell folks do not have to run either one, I think, so we tell them not to :-) 

 (`gcloud auth login` should be fine for the folks following along in a local shell, and I agree it makes more sense. It will emit a harmless error message about needing to set a project, but we do that in the next step. Should be good to go then!)